ssd_testing.py

```diff
@@ -0,0 +1,181 @@
+            # This can probably be written in a more elegant manner
+            hue = 255*i/self.num_classes
+            col = np.zeros((1,1,3)).astype("uint8")
+            col[0][0][0] = hue
+            col[0][0][1] = 128 # Saturation
+            col[0][0][2] = 255 # Value
+            cvcol = cv2.cvtColor(col, cv2.COLOR_HSV2BGR)
+            col = (int(cvcol[0][0][0]), int(cvcol[0][0][1]), int(cvcol[0][0][2]))
+            self.class_colors.append(col) 
+        
+    def run(self, video_path = 0, start_frame = 0, conf_thresh = 0.6):
+        """ Runs the test on a video (or webcam)
+        
+        # Arguments
+        video_path: A file path to a video to be tested on. Can also be a number, 
+                    in which case the webcam with the same number (i.e. 0) is 
+                    used instead
+                    
+        start_frame: The number of the first frame of the video to be processed
+                     by the network. 
+                     
+        conf_thresh: Threshold of confidence. Any boxes with lower confidence 
+                     are not visualized.
+                    
+        """
+    
+        vid = cv2.VideoCapture(video_path)
+        if not vid.isOpened():
+            raise IOError(("Couldn't open video file or webcam. If you're "
+            "trying to open a webcam, make sure you video_path is an integer!"))
+        
+        # Compute aspect ratio of video     
+        vidw = vid.get(cv2.cv.CV_CAP_PROP_FRAME_WIDTH)
+        vidh = vid.get(cv2.cv.CV_CAP_PROP_FRAME_HEIGHT)
+        vidar = vidw/vidh
+        
+        # Skip frames until reaching start_frame
+        if start_frame > 0:
+            vid.set(cv2.cv.CV_CAP_PROP_POS_MSEC, start_frame)
+            
+        accum_time = 0
+        curr_fps = 0
+        fps = "FPS: ??"
+        prev_time = timer()
+            
+        while True:
+            retval, orig_image = vid.read()
+            if not retval:
+                print("Done!")
+                return
+                
+            im_size = (self.input_shape[0], self.input_shape[1])    
+            resized = cv2.resize(orig_image, im_size)
+            rgb = cv2.cvtColor(resized, cv2.COLOR_BGR2RGB)
+            
+            # Reshape to original aspect ratio for later visualization
+            # The resized version is used, to visualize what kind of resolution
+            # the network has to work with.
+            to_draw = cv2.resize(resized, (int(self.input_shape[0]*vidar), self.input_shape[1]))
+            
+            # Use model to predict 
+            inputs = [image.img_to_array(rgb)]
+            tmp_inp = np.array(inputs)
+            x = preprocess_input(tmp_inp)
+            
+            y = self.model.predict(x)
+            
+            
+            # This line creates a new TensorFlow device every time. Is there a 
+            # way to avoid that?
+            results = self.bbox_util.detection_out(y)
```

Comment:
I can't really think of a "clean" solution here. Maybe during the creation of the bbox_util object, it could create on single TensorFlow device, which is reused every time functions are called. But I'm not sure how to write that in code, I'm not that good at TensorFlow...